river/tree/_attribute_observer/nominal_attribute_regression_observer.py

```diff
@@ -1,48 +1,54 @@
-from skmultiflow.trees._attribute_test import NominalAttributeBinaryTest
-from skmultiflow.trees._attribute_test import NominalAttributeMultiwayTest
-from skmultiflow.trees._attribute_test import AttributeSplitSuggestion
+from river.stats import Var
+from river.utils import VectorDict
+from river.tree._attribute_test import NominalAttributeBinaryTest
+from river.tree._attribute_test import NominalAttributeMultiwayTest
+from river.tree._attribute_test import AttributeSplitSuggestion
 from .attribute_observer import AttributeObserver
 
 
 class NominalAttributeRegressionObserver(AttributeObserver):
-    """ Class for observing the data distribution for a nominal attribute for
-    regression.
+    """Nominal attribute observer for regression tasks.
     """
 
     def __init__(self):
         super().__init__()
         self._statistics = {}
+        self._update_estimator = self._update_estimator_univariate
 
-    def update(self, att_val, target, weight=1.0):
-        if att_val is None or weight is None:
+    @staticmethod
+    def _update_estimator_univariate(estimator, target, sample_weight):
+        estimator.update(target, sample_weight)
+
+    @staticmethod
+    def _update_estimator_multivariate(estimator, target, sample_weight):
+        for t in target:
+            estimator[t].update(target[t], sample_weight)
+
+    def update(self, att_val, target, sample_weight=1.0):
+        if att_val is None or sample_weight is None:
             return
         else:
             try:
-                self._statistics[att_val][0] += weight
-                self._statistics[att_val][1] += weight * target
-                self._statistics[att_val][2] += weight * target * target
+                estimator = self._statistics[att_val]
             except KeyError:
-                self._statistics[att_val] = {
-                    0: weight,
-                    1: weight * target,
-                    2: weight * target * target
-                }
-                self._statistics = dict(
-                    sorted(self._statistics.items())
-                )
+                if isinstance(target, dict):  # Multi-target case
+                    self._statistics[att_val] = VectorDict(default_factory=lambda: Var())
+                    self._update_estimator = self._update_estimator_multivariate
+                else:
+                    self._statistics[att_val] = Var()
+                estimator = self._statistics[att_val]
+            self._update_estimator(estimator, target, sample_weight)
 
     def probability_of_attribute_value_given_class(self, att_val, target):
-        return 0.0
+        raise NotImplementedError
 
     def get_best_evaluated_split_suggestion(self, criterion, pre_split_dist, att_idx, binary_only):
```

Comment:
"Now is as good a time as any" XD.

I decided to address such a refactoring in the commit [f754b24](https://github.com/creme-ml/creme/pull/362/commits/f754b241b6fcce9699046135b1cf93abf89cb450) so that I make Wayne's life easier.

If you disagree with any of my renaming choices, please let me know. I basically removed the `get_` prefix from the methods.